test/errors-spec.js

```diff
@@ -0,0 +1,345 @@
+                   "Double subscript at position 9: 1_{2+3}^4̲_5");
+        });
+    });
+
+    describe("#parseImplicitGroup", function() {
+        it("reports unknown environments", function() {
+            expect("\\begin{foo}bar\\end{foo}").toFailWithParseError(
+                   "No such environment: foo at position 11:" +
+                   " \\begin{foo}̲bar\\end{foo}");
+        });
+        it("reports mismatched environments", function() {
+            expect("\\begin{pmatrix}1&2\\\\3&4\\end{bmatrix}+5")
+                .toFailWithParseError(
+                   "Mismatch: \\begin{pmatrix} matched by \\end{bmatrix}");
+        });
+    });
+
+    describe("#parseFunction", function() {
+        it("rejects math-mode functions in text mode", function() {
+            // TODO: The position info is missing here
+            expect("\\text{\\sqrt2 is irrational}").toFailWithParseError(
+                   "Can't use function '\\sqrt' in text mode");
+        });
+    });
+
+    describe("#parseArguments", function() {
+        it("complains about missing argument at end of input", function() {
+            expect("2\\sqrt").toFailWithParseError(
+                   "Expected group after '\\sqrt' at position 6: 2\\sqrt̲");
+        });
+        it("complains about missing argument at end of group", function() {
+            expect("1^{2\\sqrt}").toFailWithParseError(
+                   "Expected group after '\\sqrt' at position 9: 1^{2\\sqrt̲}");
+        });
+        it("complains about functions as arguments to others", function() {
+            // TODO: The position looks pretty wrong here
+            expect("\\sqrt\\over2").toFailWithParseError(
+                   "Got function '\\over' as argument to '\\sqrt'" +
+                   " at position 9: \\sqrt\\ove̲r2");
+        });
+    });
+
+    describe("#parseArguments", function() {
+        it("complains about missing argument at end of input", function() {
+            expect("2\\sqrt").toFailWithParseError(
+                   "Expected group after '\\sqrt' at position 6: 2\\sqrt̲");
+        });
+        it("complains about missing argument at end of group", function() {
+            expect("1^{2\\sqrt}").toFailWithParseError(
+                   "Expected group after '\\sqrt' at position 9: 1^{2\\sqrt̲}");
+        });
+        it("complains about functions as arguments to others", function() {
+            // TODO: The position looks pretty wrong here
+            expect("\\sqrt\\over2").toFailWithParseError(
+                   "Got function '\\over' as argument to '\\sqrt'" +
+                   " at position 9: \\sqrt\\ove̲r2");
+        });
+    });
+
+});
+
+describe("Parser.expect calls:", function() {
+
+    describe("#parseInput expecting EOF", function() {
+        it("complains about extra }", function() {
+            expect("{1+2}}").toFailWithParseError(
+                   "Expected 'EOF', got '}' at position 6: {1+2}}̲");
```

Comment:
It seems like the position on this one is off by one as well.  Could you add a TODO here?